docs/docsite/rst/porting_guides/porting_guide_2.10.rst

```diff
@@ -28,7 +28,7 @@ Playbook
 * A new fact, ``ansible_processor_nproc`` reflects the number of vcpus
   available to processes (falls back to the number of vcpus available to
   the scheduler).
-
+* ``conditional_bare_variables`` now defaults to ``false`` (so that ``when: result`` will always default to ``true`` if ``result`` is a non-empty string, even if that string is ``false``). See the :ref:`2.8 porting guide note <bare_variables_in_conditionals>` for more details.
```

Comment:
this is not the issue, in any case, if its already in previous porting guide, we don't need to copy in every subsequent one.

The fix that causes this is not related to 'bare' variables but to correctly typing 'boolean' keywords, which previously accepted incorrect values that ended up 'truthy' in weird ways.

Reply:
This is all semantics. If you folks want to word it better as to why my playbooks break with 2.10 but work in 2.9, be my guest.

Reply:
if you can  #67625 is what i believe is the cause, i'll write up the porting guide entry

Reply:
@bcoca: confirmed

```
- hosts: localhost
  connection: local
  vars:
    unsafe_show_logs: false

  tasks:
    - debug:
        msg: hello
      no_log: not unsafe_show_logs
```

That playbook works with `git checkout babac66f9cd978b765f58a21c20f3577308f5504^` (i.e. the commit before that one) but not with `git checkout babac66f9cd978b765f58a21c20f3577308f5504`

Reply:
@willthames  thanks, sorry for the inconvenience, I had not foreseen that many playbooks using the 'broken behaviour', which is why i had not added a porting guide entry. 

Correcting that now.

FYI, no_log was not being 'auto templated' it was just equating any strings to 'True'